crates/core/src/globals.rs

```diff
@@ -255,6 +272,60 @@ fn build_config_object(context: &JSContextRef) -> anyhow::Result<JSValueRef> {
     Ok(config_obj)
 }
 
+fn build_memory(context: &JSContextRef) -> anyhow::Result<JSValueRef> {
+    let memory_from_buffer = context.wrap_callback(
+        |_ctx: &JSContextRef, _this: JSValueRef, args: &[JSValueRef]| {
+            let data = args.get(0).ok_or(anyhow!("Expected data argument"))?;
+            if !data.is_array_buffer() {
+                bail!("Expected data to be an array buffer");
+            }
+            let data = data.as_bytes()?;
+            let m = extism_pdk::Memory::from_bytes(data)?;
+            let mut mem = HashMap::new();
+            let offset = JSValue::Int(m.offset() as i32);
+            let len = JSValue::Int(m.len() as i32);
+            mem.insert("offset".to_string(), offset);
+            mem.insert("len".to_string(), len);
+            Ok(JSValue::Object(mem))
```

Comment:
I might move some more of this logic up into the Javascript shim layer. Thinking about it. Won't affect the interface though.